packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -1,42 +1,65 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useReducer,
+  useState,
+  useEffect,
+  useRef,
+} from 'react';
+import {
+  KEY_LEFT,
+  KEY_RIGHT,
+  KEY_DELETE,
+  KEY_BACK_SPACE,
+} from 'keycode-js';
+import { injectIntl, intlShape } from 'react-intl';
 import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import uuidv4 from 'uuid/v4';
 import Button from 'terra-button';
 import IconCalendar from 'terra-icon/lib/icon/IconCalendar';
 import Input from 'terra-form-input';
-import { injectIntl, intlShape } from 'react-intl';
-import uuidv4 from 'uuid/v4';
-import classNames from 'classnames/bind';
 import ThemeContext from 'terra-theme-context';
-
+import VisuallyHiddenText from 'terra-visually-hidden-text';
 import DateUtil from './DateUtil';
 import { getLocalizedDateForScreenReader } from './react-datepicker/date_utils';
 import styles from './DatePicker.module.scss';
 
 const cx = classNames.bind(styles);
 
-const Icon = <IconCalendar />;
-
 const propTypes = {
+  /**
+   * String that labels the current element. 'aria-label' must be present for accessibility.
+   */
+  ariaLabel: PropTypes.string,
   /**
    * Callback ref to pass into the calendar button dom element.
    */
   buttonRefCallback: PropTypes.func,
+  /**
+   * The id to append to the date input wrapper.
+   */
+  id: PropTypes.string,
   /**
    * Custom input attributes to apply to the date input.
    */
   inputAttributes: PropTypes.object,
```

Comment:
Does this PR require a MVB? This discussion talks about removing `inputAttributes` in terra-time-input because it is redundant given we already have `hourAttributes`, `minuteAttributes`, and `secondAttributes`. Should we get rid of this `inputAttributes` and create `monthAttributes`, `dayAttributes`, and `yearAttributes` to be consistent. Although I don't know of a use case where consumers would have a need to pass down different attrubutes for each of these inputs.

Reply:
Should we remove those attributes all together? Do we have a scenario where consumers need to spread attributes for any of these inputs?

I'm good with a MVB with this change. it seems large enough that we might want to allow teams to opt into getting the split behaviour.

Reply:
I agree this should be an MVB.
Also i did quick internal repo search and did **not** find `inputAttributes` being used anywhere.
@benbcai @mjhenkes So do we still need to have separate spread attributes?

Reply:
I do see consumers using `inputAttributes` on DatePickerField as well on TimeInput and SearchField. Creating three separate input attribute props seems overkill. I searched and don't see any consumers using `hourAttributes`, `minuteAttributes`, or `secondAttributes` at all. So I think we should just keep `inputAttributes` despite the inconsistency with TimeInput. We can consider adding them when there is ever a request for it.

![Screen Shot 2020-10-08 at 10 51 47 AM](https://user-images.githubusercontent.com/22618655/95483103-73d25880-0954-11eb-8ced-514bd9f62b3c.png)